ovh/resource_ip_move.go

```diff
@@ -0,0 +1,309 @@
+			log.Printf("[DEBUG] Will move ip %s (service name = %s) from service %s to IP parking", ip, *serviceName, *currentlyRoutedService)
+			endpoint := fmt.Sprintf("/ip/%s/park",
+				url.PathEscape(ip),
+			)
+			// retrieve the task
+			if err := config.OVHClient.Post(endpoint, nil, ipTask); err != nil {
+				return fmt.Errorf("calling Post %s: %q", endpoint, err)
+			}
+		} else {
+			log.Printf("[DEBUG] Will move ip %s (service name = %s) from service %v to service %s", ip, *serviceName, currentlyRoutedService, *opts.To)
+			endpoint := fmt.Sprintf("/ip/%s/move",
+				url.PathEscape(ip),
+			)
+			if err := config.OVHClient.Post(endpoint, opts, ipTask); err != nil {
+				return fmt.Errorf("calling Post %s: %q", endpoint, err)
+			}
+		}
+		d.SetId(fmt.Sprint(ipTask.TaskId))
+		if err = d.Set("task_start_date", ipTask.StartDate.String()); err != nil {
+			return err
+		}
+
+		_, err = waitForTaskFinished(d, meta, ipTask, ip, opts)
+		if err != nil {
+			return err
+		}
+	}
+	err = resourceIpRead(d, meta)
+	return err
+}
+
+// waitForTaskFinished queries GET /ip/:ip/task/:taskId route until task state is in a terminal success or error state or until WaitingTimeInSecondsBeforeRefreshState is reached
+// and returns :
+//   - finishedWithSuccess : true if task ended with success, false if ended with error, nil if not ended at all
+//   - err : nil if no error is encountered in the process, any met error otherwise
+//
+// in any case before returning, "task_status" field of d will be updated with the last known ipTask.Status
+func waitForTaskFinished(d *schema.ResourceData, meta interface{}, ipTask *IpTask, ip string, opts *IpMoveOpts) (finishedWithSuccess *bool, err error) {
+	defer func() {
+		errSet := d.Set("task_status", ipTask.Status)
+		if errSet == nil {
+			err = errSet
+		}
+	}()
+	return recursiveWaitTaskFinished(d, meta, ipTask, ip, opts)
+}
+
+// recursiveWaitTaskFinished checks a given ipTask and return true if task status is in a state that we consider finished.
+// and calls itself again if task is not finished while task is not yet expired
+func recursiveWaitTaskFinished(d *schema.ResourceData, meta interface{}, ipTask *IpTask, ip string, opts *IpMoveOpts) (finished *bool, err error) {
+	switch ipTask.Status {
+	case IpTaskStatusDone:
+		return helpers.GetNilBoolPointer(true), nil
+	case IpTaskStatusCancelled, IpTaskStatusOvhError, IpTaskStatusCustomerError:
+		return helpers.GetNilBoolPointer(true), fmt.Errorf("could not assign IP %s to service %v as Ip task is %s", ip, opts.To, ipTask.Status)
+	}
+	timeDiff := time.Now().Sub(ipTask.StartDate)
+	if timeDiff < taskExpiresAfter {
+		log.Printf("[DEBUG] ipTask.Status is currently: %s. Waiting %d second (we allow %f more seconds for the task to complete)", ipTask.Status, WaitingTimeInSecondsBeforeRefreshState, taskExpiresAfter.Seconds()-timeDiff.Seconds())
+		time.Sleep(WaitingTimeInSecondsBeforeRefreshState * time.Second)
+		err = resourceIpTaskRead(d, ipTask, meta)
```

Comment:
in some instances such as a gone task it might make sense to indeed return error immediately. 

In other cases though (network error, temporary server error, etc...) stopping the whole plan is not relevant nor wanted.

Reply:
seems ok as a workaround, but IMO a better solution is to verify the error response like it's done in func `resourceIpServiceReadByServiceName` (with type assertion `err.(*ovh.APIError)` + status check)